prereise/gather/demanddata/transportation_electrification/generate_BEV_vehicle_profiles.py

```diff
@@ -0,0 +1,146 @@
+    else:
+        charging_efficiency = 0.9
+
+    geographic_area_bev_vmt = {}
+
+    urban_scaling_filepath = os.path.join(
+        const.data_folder_path,
+        "regional_scaling_factors",
+        "Regional_scaling_factors_UA_",
+    )
+    urban_scaling_factors = pd.read_csv(
+        urban_scaling_filepath + str(model_year) + ".csv", index_col=["State"]
+    )
+    state_urban_areas = urban_scaling_factors.loc[state, "UA"]
+
+    # scaling factors for listed urban areas
+    for urban_area in state_urban_areas:
+        urban_bev_vmt = load_urbanized_scaling_factor(
+            model_year=model_year,
+            veh_type=veh_type,
+            veh_range=veh_range,
+            urbanized_area=urban_area,
+            state=state,
+            filepath=urban_scaling_filepath,
+        )
+        geographic_area_bev_vmt.update({urban_area: urban_bev_vmt})
+
+    # scaling factors for rural areas
+    rural_bev_vmt = load_rural_scaling_factor(
+        model_year,
+        veh_type,
+        veh_range,
+        state,
+        filepath=os.path.join(
+            const.data_folder_path,
+            "regional_scaling_factors",
+            "Regional_scaling_factors_RA_",
+        ),
+    )
+    geographic_area_bev_vmt.update({state: rural_bev_vmt})
+
+    # calculate demand for all geographic areas with scaling factors
+    state_demand_profiles = {}
+    for geographic_area, bev_vmt in geographic_area_bev_vmt.items():
+
+        if charging_strategy == "immediate":
+            normalized_demand = immediate.immediate_charging(
+                census_region=census_region,
+                model_year=model_year,
+                veh_range=veh_range,
+                power=power,
+                location_strategy=location_strategy,
+                veh_type=veh_type,
+                filepath=os.path.join(
+                    const.data_folder_path,
+                    "nhts_census_updated_dwell.mat",
+                ),
+            )
+
+            final_demand = immediate.adjust_bev(
+                hourly_profile=normalized_demand,
+                adjustment_values=daily_values,
+                model_year=model_year,
+                veh_type=veh_type,
+                veh_range=veh_range,
+                bev_vmt=bev_vmt,
+                charging_efficiency=charging_efficiency,
+            )
+
+        elif charging_strategy == "smart":
+            final_demand = smart_charging.smart_charging(
+                census_region=census_region,
+                model_year=model_year,
+                veh_range=veh_range,
+                kwhmi=kwhmi,
+                power=power,
+                location_strategy=location_strategy,
+                veh_type=veh_type,
+                filepath=vehicle_trip_data_filepath,
+                daily_values=daily_values,
+                load_demand=load_demand,
+                bev_vmt=bev_vmt,
+                trip_strategy=trip_strategy,
+            )
```

Comment:
Are we still trying to figure out whether to call `adjust_bev` for smart charing?

Reply:
No, that is no longer needed.  The `bev_vmt` scaling is included directly in the smart charging algorithm.